### PR TITLE
Proj01

### PR DESCRIPTION
**Pull request by ClaudiaZeng**
CZ/KM W18 Proj01

Resolved issues #10, #34, #47, #64, #37, #60, #31, #59, #44, #38, #20, #32, #63

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,21 +61,26 @@ Download the lastest version of Android Studio at the following link: http://dev
 Run the app by clicking Run on Android Studio. Choose any Android Device as your emulator and the application should start. You will be prompted with a welcome screen, after which you are invited to type in your name and a welcome message will be displayed. Next screen is a simple calculator you can play around with. Following is the quiz itself, which contains checkboxes for the types of conversions (ie. Decimal to Binary or Octal to Hexadecimal) and radio buttons for the length of the quiz (ie. 6 questions, 12 questions, etc.) You will then be prompted with the quiz interface itself, where a number is given and the user will be prompted to type in the correct conversion. A "Change Number" feature is added if the current problem is too difficult. A "Percentage" feature has also been implemented to tell the user how well they are doing as they are progressing throughout the quiz. 
 
 ## Javadoc
-JavaDoc Link: http://ucsb-cs56-projects.github.io/cs56-android-conversion-showdown/javadoc/
+JavaDoc Link: https://github.com/kerrymo/cs56-android-conversion-showdown/tree/proj01/javadoc
+
+
```
Comment (rhelmot) [inline README.md:66]: Please revert this change - this is supposed to be a link to the rendered html page, not to the github file listing.

File: README.md [hunk 2]
```diff
@@ -61,21 +61,26 @@ Download the lastest version of Android Studio at the following link: http://dev
...
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss3.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss4.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss5.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss6.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss7.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss8.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss9.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss10.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss11.png)
+![Alt tag](https://github.com/kerrymo/cs56-android-conversion-showdown/blob/proj01/app/src/readmePic/ss12.png)
```
Comment (rhelmot) [inline README.md:83]: convert these links to relative instead of absolute - you can use just `/app/src/readmePic/ss#.png`

File: W18_proj00.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# Claudia Zeng Kerry Mo W18 Proj00
```
Comment (rhelmot) [inline W18_proj00.md:1]: You're missing the rest of this file... You can skip the "evaluate the code and class structure" part but you do need to do the rest of the proj00

**Review by rhelmot — CHANGES_REQUESTED** [review]
add .DS_Store to the .gitignore and delete all the .DS_Store files you've added

**Comment by rhelmot** [comment]
Also please leave a comment here with the issues you're resolving with this pull request

File: app/src/androidTest/java/edu/ucsb/cs56/projects/android/conversion_showdown/baseCalculatorTest.java [hunk 4]
```diff
@@ -96,7 +96,7 @@ public void testCalcOctal() {
     @Test
     public void testCalcHex() {
         onView(withId(R.id.firstInput))
-                .perform(typeText("F"), closeSoftKeyboard());
+                .perform(typeText("f"), closeSoftKeyboard());
```
Comment (seemantasaha) [inline app/src/androidTest/java/edu/ucsb/cs56/projects/android/conversion_showdown/baseCalculatorTest.java:99]: Make sure the project supports both F and f for hexadecimal inputs.

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/ConverterFragment.java [hunk 5]
```diff
@@ -125,16 +125,14 @@ public void onClick(View v) {
...
-                if (!converter.isValid()) {
-                    mFromText.setError("Input Format Error");
-                }
                 mToText.setText(converter.toBase(toBase));
                 MyUtils.hideSoftKeyBoard(v, inputMethodManager);
 
             } catch (IntOverFlow e) {
                 mFromText.setError("Input Number Too Large");
+            } catch (SanityCheckException e){
+                mFromText.setError("Input Format Error");
```
Comment (seemantasaha) [inline app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/ConverterFragment.java:135]: Try to make same error messages. You are using both:
"Input Format Wrong" and "Input Format Error"

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java [hunk 6]
```diff
@@ -17,19 +17,11 @@
 public class MyUtils {
 
     //The characters shown are basically what are allowed
-    final public static ArrayList<Character> symbols = new ArrayList<>( Arrays.asList(new Character[]{
-            '0', '1', '2', '3' , '4' , '5', '6' , '7' , '8' , '9',
-            'A', 'B', 'C', 'D', 'E', 'F'
-    }));
-
-    static TreeSet<Integer> baseWhiteList = new TreeSet<Integer>(){
```
Comment (seemantasaha) [inline app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java:25]: Why did you remove baseWhiteList?

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java [hunk 7]
```diff
@@ -17,19 +17,11 @@
...
-        {
-            add( 2 );
-            add( 8 );
-            add( 10 );
-            add( 16 );
-        }
-    };
+    final public static ArrayList<Character> symbols = new ArrayList<>(Arrays.asList(new Character[]{
+                '-', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
+                'A', 'B', 'C', 'D', 'E', 'F'
```
Comment (seemantasaha) [inline app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java:22]: There should be support for small letter values for hex number input.

**Review by seemantasaha — COMMENTED** [review]
When you are removing unnecessary java files make sure you are not deleting some classes which was written to use for future features. If a particular class is not being used currently it does not mean it's an unnecessary one. Make sure you looked into the code.

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java (hunk 6 above)
Comment (ClaudiaZeng) [reply app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java:25]: Because it's never used anywhere else.

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java (hunk 7 above)
Comment (ClaudiaZeng) [reply app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java:22]: It has already supported lower letters. The line 81 demonstrates how.

Comment (seemantasaha) [reply app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java:22]: Okay, line 81 supports lower letters, but not upper letters as it converts to lower case first and then matches. But, considering it for this review.

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java (hunk 6 above)
Comment (seemantasaha) [reply app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/MyUtils.java:25]: ok.

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/ConverterFragment.java (hunk 5 above)
Comment (seemantasaha) [reply app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/ConverterFragment.java:135]: Did you make these changes?

File: app/src/androidTest/java/edu/ucsb/cs56/projects/android/conversion_showdown/baseCalculatorTest.java (hunk 4 above)
Comment (seemantasaha) [reply app/src/androidTest/java/edu/ucsb/cs56/projects/android/conversion_showdown/baseCalculatorTest.java:99]: I was asking to add support for both small and capital letters for hex value.

**Comment by seemantasaha** [comment]
@ClaudiaZeng and @kerrymo ,
Would you please fix the asked changes and come to the Friday open lab between 3-5 pm so that I can check the application after running and talk to you about the review comments I made. After that, I can merge this pull request.

File: app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/ConverterFragment.java (hunk 5 above)
Comment (kerrymo) [reply app/src/main/java/edu/ucsb/cs56/projects/android/conversion_showdown/ConverterFragment.java:135]: Yes we did fix it and push to the branch proj02

File: app/src/androidTest/java/edu/ucsb/cs56/projects/android/conversion_showdown/baseCalculatorTest.java (hunk 4 above)
Comment (kerrymo) [reply app/src/androidTest/java/edu/ucsb/cs56/projects/android/conversion_showdown/baseCalculatorTest.java:99]: Fixed

File: README.md (hunk 1 above)
Comment (kerrymo) [reply README.md:66]: Fixed

File: README.md (hunk 2 above)
Comment (kerrymo) [reply README.md:83]: Fixed
